server/storages/s3.go

```diff
@@ -80,9 +80,8 @@ func (s3 *S3) Store(fileName string, objects []map[string]interface{}, alreadyUp
 
 	storeFailedEvents := true
 	tableResults := map[string]*StoreResult{}
-	marshaller := s3.marshaller()
 	for _, fdata := range processedFiles {
-		b := fdata.GetPayloadBytes(marshaller)
+		b := s3.marshall(fdata)
```

Comment:
let's refactor this code to:
1. create a func like: func load (f *schema.ProcessedFile) error{
                b,err := s3.marshall(fdata)
                if err != nil { 
                  return err
                }
		fileName := s3.fileName(fdata)
		return s3.s3Adapter.UploadBytes(fileName, b)
}

2. replace this code with:
for _, fdata := range processedFiles {
err := s3.load(fdata)
tableResults[fdata.BatchHeader.TableName] = &StoreResult{Err: err, RowsCount: fdata.GetPayloadLen(), EventsSrc: fdata.GetEventsPerSrc()}
		if err != nil {
			logging.Errorf("[%s] Error storing file %s: %v", s3.ID(), fileName, err)
			storeFailedEvents = false
		}
		//other code as is
		....

}

Reply:
done